app/models/concerns/urgency.rb

```diff
@@ -4,26 +4,26 @@ module Urgency
 
   def still_urgent?
     # Verify that a pregnancy has not been marked urgent in the past six days
-    return false if recent_history_tracks.count == 0
+    return false if recent_history_tracks.count.zero?
     return false if pledge_sent || resolved_without_fund
     recent_history_tracks.sort.reverse.each do |history|
       return true if history.marked_urgent?
     end
     false
   end
 
+  def confirm_still_urgent
+    update urgent_flag: false unless still_urgent?
```

Comment:
Here's the 🐛

Reply:
Oh, I think this was infinitely triggering the after_save callback somehow. Adding a conditional to check urgent status seems to fix it.